packages/create-subscription/src/createSubscription.js

```diff
@@ -122,15 +125,14 @@ export function createSubscription<Property, Value>(
         // This is safe to do via mutation since:
         // 1) It does not impact render.
         // 2) This method will only be called during the "commit" phase.
-        const unsubscribeOrBoolean = subscribe(source, callback);
+        const unsubscribe = subscribe(source, callback);
 
         invariant(
-          unsubscribeOrBoolean === false ||
-            typeof unsubscribeOrBoolean === 'function',
+          typeof unsubscribe === 'function',
```

Comment:
Does this need to be an invariant, as opposed to a warning?

Reply:
Yeah, why not? This relates to my previous comment. https://github.com/facebook/react/pull/12325#discussion_r174232329